fn/goroutine_manager_test.go

```diff
@@ -0,0 +1,109 @@
+package fn
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestGoroutineManager tests that GoroutineManager starts goroutines, until ctx
+// expires fails to start after it expires and waits for already started
```

Comment:
Nit: // TestGoroutineManager tests that the GoroutineManager starts goroutines until ctx
// expires. It also makes sure it fails to start new goroutines after the context expired and the GoroutineManager is in the process of waiting for already started goroutines in the Stop method.

Reply:
Done. Thanks!